rs/registry/canister/src/mutations/node_management/do_replace_operator.rs

```diff
@@ -0,0 +1,829 @@
+                )]
+                .into_iter()
+                .collect(),
+            );
+        mutations.extend(threshold_pk_and_cup_mutations);
+
+        // One more node is required for
+        for node in nodes.iter().chain(&[&one_more_nns_node]) {
+            mutations.extend(vec![
+                key_mutation(
+                    node.node_id(),
+                    node.node_signing_key().key_value.clone(),
+                    KeyPurpose::NodeSigning,
+                ),
+                key_mutation(
+                    node.node_id(),
+                    node.committee_signing_key().key_value.clone(),
+                    KeyPurpose::CommitteeSigning,
+                ),
+                key_mutation(
+                    node.node_id(),
+                    node.dkg_dealing_encryption_key().key_value.clone(),
+                    KeyPurpose::DkgDealingEncryption,
+                ),
+                key_mutation(
+                    node.node_id(),
+                    node.idkg_dealing_encryption_key().key_value.clone(),
+                    KeyPurpose::IDkgMEGaEncryption,
+                ),
+                certificate_mutation(node.node_id(), node.tls_certificate().clone()),
+                unassigned_nodes_record(),
+            ]);
+        }
+
+        mutations
+    }
+
+    fn routing_table_mutation() -> RegistryMutation {
+        let routing_table = RoutingTable {
+            entries: vec![Entry {
+                range: Some(CanisterIdRange {
+                    start_canister_id: Some(CanisterId::from(0).into()),
+                    end_canister_id: Some(CanisterId::from(100).into()),
+                }),
+                subnet_id: Some(SubnetId {
+                    principal_id: Some(PrincipalId::new_subnet_test_id(0).into()),
+                }),
+            }],
+        };
+
+        mutation(
+            make_routing_table_record_key().as_bytes().to_vec(),
+            routing_table.encode_to_vec(),
+        )
+    }
+
+    fn generate_valid_node_id() -> ValidNodePublicKeys {
+        let (config, _tmp_dir) = CryptoConfig::new_in_temp_dir();
+        generate_node_keys_once(&config, None).unwrap()
+    }
+
+    fn key_mutation(node: NodeId, key: Vec<u8>, purpose: KeyPurpose) -> RegistryMutation {
+        let key = PublicKey {
+            version: 0,
+            algorithm: AlgorithmId::Ed25519 as i32,
+            key_value: key,
+            proof_data: None,
+            timestamp: Some(42),
+        };
+
+        mutation(
+            make_crypto_node_key(node, purpose).as_bytes().to_vec(),
+            key.encode_to_vec(),
+        )
+    }
+
+    fn certificate_mutation(node_id: NodeId, certificate: X509PublicKeyCert) -> RegistryMutation {
+        mutation(
+            make_crypto_tls_cert_key(node_id).as_bytes().to_vec(),
+            certificate.encode_to_vec(),
+        )
+    }
+
+    fn nns(node_id: NodeId) -> RegistryMutation {
```

Comment:
upsert_nns_subnet_record

Reply:
Feel's too verbose and we are not going to have "delete's".
Also it is just to make an invariant-compliant registry, not at all related to the actual tested code

Reply:
Function has been removed.